shells/lib/components/arc-host.js

```diff
@@ -16,10 +16,11 @@ import {Utils} from '../runtime/utils.js';
 const {log, warn, error} = logsFactory('ArcHost', '#cade57');
 
 export class ArcHost {
-  constructor(context, storage, composer) {
+  constructor(context, storage, composer, portFactories) {
```

Comment:
Syntax is increasingly nasty, will have to consider creating some kind of 'arc-deps' structure instead of exploding all the possibilities into arguments.

Also, at some point I'd like to discuss `portFactories` so I can understand better why we need that level of flexibility here.